app/Http/Controllers/CaasController.php

```diff
@@ -3,70 +3,168 @@
 namespace App\Http\Controllers;
 
 use App\Models\Announcecheck;
+use App\Models\Assistantcontact;
 use App\Models\Datacaas;
 use App\Models\Grouprole;
 use App\Models\Message;
 use App\Models\Plotting;
 use App\Models\Roles;
 use App\Models\Shift;
 use App\Models\Status;
+use Carbon\Carbon;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
 
 class CaasController extends Controller
 {
     public function index()
     {
-        $profile = Datacaas::all();
+        $path = Auth::guard('caas')->user()->role->roles->character_photo;
         $data = [
-            'title' => "Caas Dashboard",
-            'profile' => $profile,
+            'title' => "Explorer's Dashboard",
+            'path' => $path
         ];
-        return view('caas.dashboard', $data);
+        return view('caas.index', $data);
     }
-    public function checkStatus()
+    public function announcement()
     {
-        $status = Status::first();
-        return view('status.dashboard', ['status' => $status]);
+        $msg = Message::first();
+        $data = [
+            'title' => "Pengumumman",
+            'msg' => $msg
+        ];
+        return view('caas.announcement', $data);
     }
+
+    public function assistant()
+    {
+        $ass = Assistantcontact::all();
+        $data = [
+            'title' => "Kontak Asisten",
+            'assistant' => $ass
+        ];
+        return view('caas.assistant', $data);
+    }
+
+    //Schedule functions
     public function viewSchedule()
     {
-        $shifts = Shift::all();
-        return view('shift.caas', compact('shifts'));
+        if (Auth::guard('caas')->user()->status->isPass == 0 || Auth::guard('caas')->user()->announcecheck()->isPlotActive == 0) {
+            return redirect()->route('caas.dashboard');
+        } elseif (boolval(Auth::guard('caas')->user()->plots) == 1) {
+            return redirect()->route('caas.fix.schedule');
+        }
+        $shifts = Shift::orderBy('day', 'asc')->orderBy('start_hour', 'asc')->paginate(5);
+        $data = [
+            'title' => "Daftar Jadwal",
+            'schedule' => $shifts
+        ];
+        return view('caas.schedule', $data);
     }
+
     public function selectSchedule(Request $request)
     {
-        $request->validate([
-            'shift' => 'required|exists:shifts,id',
-        ]);
-        $selectedShift = Shift::find($request->shift);
-        $plotting = new Plotting([
+        $shift = Shift::where('id', $request->id)->first();
+        $day = Carbon::parse($shift->day)->locale('id')->translatedFormat('l d M');
+        //kalo ada yg iseng masuk lewat link
+        if ($shift->quota <= 0 || Auth::guard('caas')->user()->announcecheck()->isPlotActive == 0) {
```

Comment:
```suggestion
        if ($shift->quota <= 0 || !Auth::guard('caas')->user()->announcecheck()->isPlotActive) {
```

same as the other one